tests/playwright/Report.md

```diff
@@ -0,0 +1,267 @@
+# Evaluation of Microsoft Playwright
+
+## Table of Contents
+
+- [Introduction](#Introduction)
+
+- [Project Background](#project-background)
+
+- [WebdriverIO](#webdriverIO)
+  - [Sample WebdriverIO test code](#sample-playwright-test-code)
+    - [Mocha](#webdriverIO-and-mocha)
+      - [WebdriverIO and Mocha code](#sample-webdriverIO-and-mocha-code)
+- [Playwright](#playwright)
+  - [Sample Playwright test code](#sample-playwright-test-code)
+  - [Jest](#playwright-and-jest)
+    - [Jest configuration](#example-of-jest-playwright-config-js)
+- [Playwright Vs WebdriverIO](#playwright-vs-webdriverIO)
+  - [Metrics](#playwright-and-webdriverIO-metrics)
+    - [Statistics](#statistics-on-webdriverIO-and-playwright-popularity)
+    - [MediaWiki Core Test Suite](#webdriverIO-and-playwright-mediawiki-core-tests-analysis)
+  - [Why adapt Playwright](#why-adapt-playwright)
+  - [Migrating from WebdriverIO](#advantages-and-disadvantages-of-migrating-from-webdriverI)
+    - [Advantages](#advantages)
+    - [Disadvantages](#disadvantages)
+
+- [What Next](#what-next)
+
+- [Summary](#summary)
+
+- [Acknowledgement](#acknowledgement)
+
+- [References](#references)
+
+## Introduction
+
+Test automation is a way to automate one's browser by simulating user actions like clicks. Applications become more complex as they are built on layers and entire networks of sub-systems, including UI and API layers, external databases, networks, and even third-party integrations. So, there is always a need for thorough testing to be done and this ranges from unit testing to end-to-end testing of applications. Test automation is one of the ways to ascertain the stability and the health of applications which can lead to the success of the application when on production. The general stability of the application doesn't entirely depend on end-to-end testing but it surely helps to detect bugs in applications, among other benefits.
+
+Below are some of the popular test automation frameworks being used;
+
+- Selenium
+- Puppeteer
+- Cypress
+- Playwright
+
+Playwright maybe the new kid on the block but it's gaining quite a trajectory among Software developers and testers.
+
+## Project Background
+
+The Playwright evaluation is a result of a need for to check if [WebdriverIO](https://webdriver.io/) is still a good test automation framework compared to some of the best non-selenium modern test automation framework. Wikimedia supports a large number of repositories. So, in order to ensure good code practices across all these repositories, an extensive amount of testing is performed and one of the tests performed is an end to end test. WebdriverIO is the current browser automation framework being used for implementing the end-to-end tests. However, with the recent increase in awareness about end-to-end testing, a number of equally competitive non-selenium solutions have been introduced and one of them is [Playwright](https://github.com/microsoft/playwright).
```

Comment:
```
Wikimedia supports a large number of repositories.
```

This doesn't sound good to me. Wikimedia is the movement, it doesn't support repositories. Maybe a better statement would be

```
MediaWiki is implemented in a large number of repositories.
```

Meaning, MediaWiki (the software) consists of Core and many extensions and skins. Is that what you wanted to say here?